File: EASY/conversor unidades/codigo.rb
def convertir_longitud(valor, de, a)
    conversiones = {
      "metros" => 1.0,
      "kilometros" => 1000.0,
      "millas" => 1609.34,
      "yardas" => 0.9144
    }
    valor * conversiones[a] / conversiones[de]
  end
  
  def convertir_masa(valor, de, a)
    conversiones = {
      "gramos" => 1.0,
      "kilogramos" => 1000.0,
      "libras" => 453.592,
      "onzas" => 28.3495
    }
    valor * conversiones[a] / conversiones[de]
  end
  
  def convertir_temperatura(valor, de, a)
    if de == "Celsius" && a == "Fahrenheit"
      valor * 9 / 5 + 32
    elsif de == "Fahrenheit" && a == "Celsius"
      (valor - 32) * 5 / 9
    elsif de == "Celsius" && a == "Kelvin"
      valor + 273.15
    elsif de == "Kelvin" && a == "Celsius"
      valor - 273.15
    elsif de == "Fahrenheit" && a == "Kelvin"
      (valor - 32) * 5 / 9 + 273.15
    elsif de == "Kelvin" && a == "Fahrenheit"
      (valor - 273.15) * 9 / 5 + 32
    else
      valor
    end
  end
  
  def conversor
    puts "Bienvenido al Conversor de Unidades"
    puts "Seleccione la categoría: 1. Longitud 2. Masa 3. Temperatura"
    categoria = gets.chomp.to_i
  
    puts "Ingresa el valor a convertir: "
    valor = gets.chomp.to_f
  
    case categoria
    when 1
      puts "Seleccione las unidades de Longitud: metros, kilometros, millas, yardas"
      puts "De: "
      de = gets.chomp.downcase
      puts "A: "
      a = gets.chomp.downcase
      puts "Resultado: #{convertir_longitud(valor, de, a)} #{a}"
    when 2
      puts "Seleccione las unidades de Masa: gramos, kilogramos, libras, onzas"
      puts "De: "
      de = gets.chomp.downcase
      puts "A: "
      a = gets.chomp.downcase
      puts "Resultado: #{convertir_masa(valor, de, a)} #{a}"
    when 3
      puts "Seleccione las unidades de Temperatura: Celsius, Fahrenheit, Kelvin"
      puts "De: "
      de = gets.chomp
      puts "A: "
      a = gets.chomp
      puts "Resultado: #{convertir_temperatura(valor, de, a)} #{a}"
    else
      puts "Opción no válida"
    end
  end
  
  conversor
  

File: EASY/conversor unidades/codigo.cpp
#include <iostream>
#include <string>
#include <map>
using namespace std;

double convertir_longitud(double valor, string de, string a) {
    map<string, double> conversiones = {
        {"metros", 1.0},
        {"kilometros", 1000.0},
        {"millas", 1609.34},
        {"yardas", 0.9144}
    };
    return valor * conversiones[a] / conversiones[de];
}

double convertir_masa(double valor, string de, string a) {
    map<string, double> conversiones = {
        {"gramos", 1.0},
        {"kilogramos", 1000.0},
        {"libras", 453.592},
        {"onzas", 28.3495}
    };
    return valor * conversiones[a] / conversiones[de];
}

double convertir_temperatura(double valor, string de, string a) {
    if (de == "Celsius" && a == "Fahrenheit")
        return valor * 9 / 5 + 32;
    if (de == "Fahrenheit" && a == "Celsius")
        return (valor - 32) * 5 / 9;
    if (de == "Celsius" && a == "Kelvin")
        return valor + 273.15;
    if (de == "Kelvin" && a == "Celsius")
        return valor - 273.15;
    if (de == "Fahrenheit" && a == "Kelvin")
        return (valor - 32) * 5 / 9 + 273.15;
    if (de == "Kelvin" && a == "Fahrenheit")
        return (valor - 273.15) * 9 / 5 + 32;
    return valor;
}

void conversor() {
    cout << "Bienvenido al Conversor de Unidades" << endl;
    cout << "Seleccione la categoria: 1. Longitud 2. Masa 3. Temperatura" << endl;
    int categoria;
    cin >> categoria;

    cout << "Ingresa el valor a convertir: ";
    double valor;
    cin >> valor;

    string de, a;
    switch (categoria) {
        case 1:
            cout << "Seleccione las unidades de Longitud: metros, kilometros, millas, yardas" << endl;
            cout << "De: ";
            cin >> de;
            cout << "A: ";
            cin >> a;
            cout << "Resultado: " << convertir_longitud(valor, de, a) << " " << a << endl;
            break;
        case 2:
            cout << "Seleccione las unidades de Masa: gramos, kilogramos, libras, onzas" << endl;
            cout << "De: ";
            cin >> de;
            cout << "A: ";
            cin >> a;
            cout << "Resultado: " << convertir_masa(valor, de, a) << " " << a << endl;
            break;
        case 3:
            cout << "Seleccione las unidades de Temperatura: Celsius, Fahrenheit, Kelvin" << endl;
            cout << "De: ";
            cin >> de;
            cout << "A: ";
            cin >> a;
            cout << "Resultado: " << convertir_temperatura(valor, de, a) << " " << a << endl;
            break;
        default:
            cout << "Opción no válida" << endl;
    }
}

int main() {
    conversor();
    return 0;
}
